### PR TITLE
fix: allow duplicate properties in block

### PR DESCRIPTION
**Pull request by pancaspe87**
Fix for https://github.com/atlassian-labs/compiled/issues/1112

### CODE REVIEW

**Review by at-nathan — APPROVED** [review]

File: packages/babel-plugin/src/utils/manipulate-template-literal.ts [hunk 1]
```diff
@@ -133,9 +133,10 @@ export const moveCssPropertyInExpression = (
   // Remove CSS property so it doesn't get processed again
   if (deleteValue) {
     let nextQuasisNewCSS = nextQuasisExtractedCss.css;
+    const newValue = value.substring(0, value.lastIndexOf(before));
```
Comment (at-nathan) [inline packages/babel-plugin/src/utils/manipulate-template-literal.ts:136]: Does this also work if the expression is before the duplicate?
i.e.
```
font-weight: ${(props) => (props.isBolded ? 'bold' : props.isPrimary ? 'normal' : 'light')};
font-weight: initial;
```

**Comment by at-nathan** [comment]
Even with this fix the new conditional evaluation is failing on expressions such as:
```
height: calc(
  ${({ customHeight }) => customHeight != null ?
    `100vh - ${PAGE_LAYOUT_OFFSET_TOP} - ${gridSize * 4 + gridSize * 8}px`
});
```
```
border-color: ${({ color }) => (color && COLOR_MAP[color] ? COLOR_MAP[color] : colors.B400)};
```

File: packages/babel-plugin/src/utils/manipulate-template-literal.ts (hunk 1 above)
Comment (pancaspe87) [reply packages/babel-plugin/src/utils/manipulate-template-literal.ts:136]: Hey @JakeLane thanks for adding a story for it! :)

@najjiatlassian yes it does work with the above case.` font-weight: initial;` should be the nextQuasis ( the next template element); hence, it is not affected by this fix

**Comment by pancaspe87** [comment]
Hey @at-nathan re https://github.com/atlassian-labs/compiled/pull/1113#issuecomment-1058785012
I wonder if those issues are related to other known issues such as
- https://github.com/atlassian-labs/compiled/issues/848
- groups of composable styles

File: packages/babel-plugin/src/utils/manipulate-template-literal.ts (hunk 1 above)
Comment (at-nathan) [reply packages/babel-plugin/src/utils/manipulate-template-literal.ts:136]: I see, just trying to make sure CSS order doesn't get changed.

**Comment by at-nathan** [comment]
@pancaspe87 Sorry I entered the expression wrongly. It has no default value, I've updated the comment. It's possibly caused by the `calc(...)`.

**Comment by JakeLane** [comment]
> @pancaspe87 Sorry I entered the expression wrongly. It has no default value, I've updated the comment. It's possibly caused by the `calc(...)`.

I think the calc issue already exists - we should probably raise an issue for it. From my understanding, unitless (e.g. number types) sizes do not get the implicit `px` added and cause invalid CSS.

No luck reproducing the failure on the border-color issue

**Comment by at-nathan** [comment]
I've put up another PR that will address the current problems introduced with conditional expression optimization. It will also fix the problem this PR is addressing
https://github.com/atlassian-labs/compiled/pull/1152

**Comment by at-nathan** [comment]
Closing this as we just merged
https://github.com/atlassian-labs/compiled/commit/2ad385cfadd0a8fb3021b3777d402acc6b9c2eea

**Pull request closed without merging**